apis/s3/src/main/java/org/jclouds/s3/filters/RequestAuthorizeSignature.java

```diff
@@ -68,191 +68,192 @@
 /**
  * Signs the S3 request.
  */
-@Singleton
-public class RequestAuthorizeSignature implements HttpRequestFilter, RequestSigner {
 
-   private static final Collection<String> FIRST_HEADERS_TO_SIGN = ImmutableList.of(HttpHeaders.DATE);
+public interface RequestAuthorizeSignature extends HttpRequestFilter {
+    @Singleton
+    public static class RequestAuthorizeSignatureV2 implements RequestAuthorizeSignature, RequestSigner {
+        private static final Collection<String> FIRST_HEADERS_TO_SIGN = ImmutableList.of(HttpHeaders.DATE);
 
-   private static final Set<String> SIGNED_PARAMETERS = ImmutableSet.of("acl", "torrent", "logging", "location", "policy",
+        private static final Set<String> SIGNED_PARAMETERS = ImmutableSet.of("acl", "torrent", "logging", "location", "policy",
```

Comment:
Right, but can we move this code into its own source file instead of it having a inner class in the interface?